rest/route/middleware.go

```diff
@@ -631,7 +631,7 @@ func urlVarsToProjectScopes(r *http.Request) ([]string, int, error) {
 
 	projectRef, err := model.FindMergedProjectRef(projectID)
 	if err != nil {
-		return nil, http.StatusInternalServerError, errors.WithStack(err)
+		return nil, http.StatusNotFound, errors.WithStack(err)
```

Comment:
I actually have PR up (although I need to fix a test) to change FindMergedProjectRef to return `nil, nil` if not found, instead of an error, which is more consistent. In that case, this _should_ be http.StatusInternalServerError, but since that PR is going to take some time I can just change this back in that PR once this is merged.